### PR TITLE
Support arbitrary parameters in visualization

### PR DESCRIPTION
**Pull request by princjef**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@princjef__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (jepri). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: DeviceAdministration/Web/Scripts/Views/Dashboard/TelemetryHistory.js [hunk 1]
```diff
@@ -11,6 +11,13 @@
         var targetControlTitle;
         var visual;
         var width;
+        var hasVisualBeenInitialized = false;
+        
+        var reservedDataColumns = [
+            "timestamp",
+            "deviceId",
+            "externalTemperature" // TODO: remove this
```
Comment (princjef) [inline DeviceAdministration/Web/Scripts/Views/Dashboard/TelemetryHistory.js]: Currently, we get back `externalTemperature` with the data returned from the API, but it's null.

File: DeviceAdministration/Web/Scripts/Views/Dashboard/TelemetryHistory.js [hunk 2]
```diff
@@ -46,42 +53,39 @@
...
+                }
+            ];
+            
+            columns = [];
+            
+            // Create a new column for non-reserved values
+            for (var field in data[0]) {
+                if (reservedDataColumns.indexOf(field) === -1) {
+                    graphMetadataColumns.push({
+                        displayName: field,
```
Comment (princjef) [inline DeviceAdministration/Web/Scripts/Views/Dashboard/TelemetryHistory.js]: The `displayName` will match the field name unless there is some way for the fields to specify their friendly name

File: DeviceAdministration/Web/Scripts/Views/Dashboard/TelemetryHistory.js [hunk 3]
```diff
@@ -135,7 +139,7 @@
             var width;
 
             pluginService = powerbi.visuals.visualPluginFactory.create();
-            singleVisualHostServices = powerbi.visuals.singleVisualHostServices;
+            singleVisualHostServices = powerbi.visuals.defaultVisualHostServices;
```
Comment (princjef) [inline DeviceAdministration/Web/Scripts/Views/Dashboard/TelemetryHistory.js]: Upon inspection of the `powerbi.visuals` object in a web browser, there is no `singleVisualHostServices` property, but there is a `defaultVisualHostServices` one. Changing the property fixes a `this.hostServices` is undefined error in the browser that prevents the visualization from loading.

File: DeviceAdministration/Web/Scripts/Views/Dashboard/TelemetryHistory.js [hunk 4]
```diff
@@ -215,7 +228,7 @@
             height = $(targetControl).height();
             width = $(targetControl).width();
 
-            if (lastData) {
+            if (lastData && hasVisualBeenInitialized) {
```
Comment (princjef) [inline DeviceAdministration/Web/Scripts/Views/Dashboard/TelemetryHistory.js]: It seems like the visual is sometimes updated before it's created for the first time. This ensures that doesn't happen

**Comment by jamesweb-ms** [comment]
old pr ... closing

**Pull request closed without merging**